src/common/http/http.cpp

```diff
@@ -115,11 +115,24 @@ error::Error BreakDownUrl(const string &url, BrokenDownUrl &address) {
 		address.path = tmp.substr(split_index);
 	}
 
-	if (address.host.find("@") != string::npos) {
-		address = {};
-		return error::Error(
-			make_error_condition(errc::not_supported),
-			"URL Username and password is not supported");
+	auto auth_index = address.host.find("@");
+	if (auth_index != string::npos) {
+		if (!with_auth) {
+			address = {};
+			return error::Error(
+				make_error_condition(errc::not_supported),
+				"URL Username and password is not supported");
+		}
+		auto user_password = address.host.substr(0, auth_index);
+		address.host = address.host.substr(auth_index + 1);
+		auto u_pw_sep_index = user_password.find(":");
+		if (u_pw_sep_index == string::npos) {
+			// no password
+			address.username = std::move(user_password);
+		} else {
+			address.username = user_password.substr(0, u_pw_sep_index);
+			address.password = user_password.substr(u_pw_sep_index + 1);
+		}
```

Comment:
Should we unescape the `%` escape sequences for the username/password?

Reply:
As discussed in the SU, this is tricky and goes against the special handling of `@` in the username/password. Either we treat it as encoded and `@` should never show up there or it is not encoded and we should not treat things like `%20` in any special way. But let's try to be as much backwards compatible as possible with the Go client and do both.